Libraries/xcassets/Headers/xcassets/Asset/Asset.h

```diff
@@ -29,6 +29,7 @@ class Asset {
 private:
     FullyQualifiedName         _name;
     std::string                _path;
+    libutil::Filesystem const *_filesystem;
```

Comment:
Hm, this is a little weird. First, though, I agree that it makes sense that the same filesystem should necessarily be the one used to read it later.

However... this does create a sort of strange situation in two ways. First, a data object is holding onto a resource, which conceptually makes the object more complex than just being immutable data. Second, the lifetime of the filesystem is unspecified in the API, and it's up to clients of this class to keep the filesystem alive at least as long as the asset is, which is definitely non-obvious. Rust's lifetimes could enforce that at compile time, but lacking those in C++ I think it's better to avoid the complexity and pass around the filesystem separately where it's needed.